src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +240,52 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                {
+                    if (test.Properties.ContainsKey(PropertyNames.Order))
+                    {
+                        Children.Insert(0, WorkItem.CreateWorkItem(test, _childFilter));
+                        _countOrder++;
+                    }
+                    else
+                    {
+                        Children.Insert(Children.Count, WorkItem.CreateWorkItem(test, _childFilter));
```

Comment:
This inserting ordered items at the beginning of the list and non-ordered items to the end works, but feels like premature optimization to me. The [underlying implementation of List<T>](http://referencesource.microsoft.com/#mscorlib/system/collections/generic/list.cs,cf7f4095e4de7646) is an array, not a linked list, so repeatedly inserting at the beginning is likely more expensive than the eventual sort since it requires constant reallocations and array copies.